includes/fbbot/class-omise-fbbot-payment-handler.php

```diff
@@ -0,0 +1,253 @@
+		global $payment_page_detect;
+    
+    $payment_page_url = "pay-on-messenger";
+    $payment_purchase_complete = "complete-payment";
+    $payment_error_url = "pay-on-messenger-error";
+
+    if ( strtolower( $wp->request ) == $payment_purchase_complete ) {
+      $image_url = site_url() . '/wp-content/plugins/omise-woocommerce/assets/images/omise_logo.png';
+      $url = 'https://www.messenger.com/closeWindow/?image_url=' . $image_url . '&display_text=THANKS%20FOR%20PURCHASE';
+      if ( wp_redirect( $url ) ) {
+        exit;
+      }
+
+      return $posts;
+    }
+
+    // Create custom page
+    $post = new stdClass;
+
+    if ( strtolower( $wp->request ) == $payment_page_url ) {
+      $post->post_title = __( 'Your order' );
+      $post->post_content = $this->payment_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_error_url ) {
+      $post->post_title = __( 'System error' );
+      $post->post_content = $this->payment_error_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_page_url || strtolower( $wp->request ) == $payment_error_url ) {
+      // Create cumtom page content
+      $post->post_author = 1;
+      $post->post_name = strtolower( $wp->request );
+      $post->guid = get_bloginfo( 'wpurl' ) . '/' . strtolower( $wp->request );
+      $post->ID = -1;
+      $post->post_type = 'page';
+      $post->post_status = 'static';
+      $post->comment_status = 'closed';
+      $post->ping_status = 'open';
+      $post->comment_count = 0;
+      $post->post_date = current_time( 'mysql' );
+      $post->post_date_gmt = current_time( 'mysql', 1 );
+      $posts = NULL;
+      $posts[] = $post;
+      
+      // make wp_query
+      $wp_query->is_page = true;
+      $wp_query->is_singular = true;
+      $wp_query->is_home = false;
+      $wp_query->is_archive = false;
+      $wp_query->is_category = false;
+      unset( $wp_query->query["error"] );
+      $wp_query->query_vars["error"] = "";
+      $wp_query->is_404 = false;
+    }
+
+    return $posts;
+	}
+
+	private function payment_page_render () {
+		global $wp_query;
+
+  	if ( ! isset( $wp_query->query_vars['product_id'] ) || ! isset( $wp_query->query_vars['messenger_id'] ) ) {
+  		return '<strong>404 Your order not found</strong>';
+  	}
+
+  	$product_id = $wp_query->query_vars['product_id'];
+  	$messenger_id = $wp_query->query_vars['messenger_id'];
+
+    $url = plugin_dir_path(dirname(__DIR__)) . 'templates/fbbot/payment-form.php';
+    ob_start();
+    include( $url );
+    return ob_get_clean();
+	}
+
+  private function payment_error_page_render () {
+    global $wp_query;
+
+    if ( ! isset( $wp_query->query_vars['error_message'] ) ) {
+      return '<strong>Woocommerce system has error. Please try again.</strong>';
+    }
+
+    $error_message = $wp_query->query_vars['error_message'];
+
+    return '<strong>' . $error_message . '</strong>';
+  }
+
+  public function process_payment_by_bot( $params ) {
+    /* rearrange step
```

Comment:
@igroomgrim How about to describe this one at https://github.com/omise/omise-woocommerce/pull/58/files#diff-b0e25579a2040d94b4d78f2b56776d79R135 by code?

I meant, this method has to do many tasks in just one method. Might be better to reduce its tasks to just only focusing on 'process a payment' and move other tasks into the above link? Then, you don't need to leave note like this anymore since this one just only perform 1 task, short and sweet 😁